cmd/create/breakglasscredential/cmd.go

```diff
@@ -0,0 +1,77 @@
+package breakglasscredential
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/breakglasscredential"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var breakGlassCredentialArgs *breakglasscredential.BreakGlassCredentialArgs
+
+const argsPrefix string = ""
+
+var Cmd = &cobra.Command{
+	Use:     "break-glass-credential",
+	Aliases: []string{"break-glass-credentials", "breakglasscredential", "breakglasscredentials"},
+	Short:   "Create a break glass credential for a cluster.",
+	Long:    "Create a break glass credential for a hosted control plane cluster with external OIDC.",
+	Example: `  # Interactively create a break glass credential to a cluster named "mycluster"
+  rosa create break-glass-credential --cluster=mycluster --interactive`,
+	Run:    run,
+	Hidden: true,
+	Args:   cobra.NoArgs,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddClusterFlag(Cmd)
+
+	interactive.AddFlag(flags)
+	breakGlassCredentialArgs = breakglasscredential.AddBreakGlassCredentialFlags(Cmd, argsPrefix)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+	err := runWithRuntime(r, cmd, argv)
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func runWithRuntime(r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	if !interactive.Enabled() {
+		interactive.Enable()
+		r.Reporter.Infof("Enabling interactive mode")
+	}
+	r.Reporter.Debugf("Creating a break glass credential for cluster '%s'", clusterKey)
+
+	// need to check if cluster is in ready state. I wonder if this should be the same as IsExternalAuthProviderSupported
```

Comment:
I would check:
- if hypershift cluster
- if ready cluster
- external auth?
All of this are also done in the backend but we usually check it before calling it

Reply:
Used a predefined function IsExternalAuthProviderSupported that checks all three